### PR TITLE
Feat!: Remove env variable that disabled setting resources for local ray

### PR DESCRIPTION
**Pull request by SebastianMorawiec**
# The problem

It was confusing for users why task resources were not used in local ray

# This PR's solution

Always set resources on local ray runtime

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/internal/docs>[!]:`

### CODE REVIEW

**Comment by SebastianMorawiec** [comment]
cc @VolodyaCO

**Review by alexjuda — APPROVED** [review]
LGTM, but let's wait for @jamesclark-Zapata's opinion before we merge this.

**Review by jamesclark-Zapata — APPROVED** [review]

**Pull request merged**
